sale_warehouse_calendar/tests/test_sale_order_dates.py

```diff
@@ -0,0 +1,124 @@
+CUTOFF_TIME = 8.0  # cutoff time will be set at 8 a.m.
+
+FRIDAY = "2021-08-13"
+SATURSDAY = "2021-08-14"
+NEXT_MONDAY = "2021-08-16"
+NEXT_TUESDAY = "2021-08-17"
+FRIDAY_BEFORE_CUTOFF = "2021-08-13 07:00:00"
+FRIDAY_AFTER_CUTOFF = "2021-08-13 09:00:00"
+
+
+class TestSaleOrderDates(SavepointCase):
+    at_install = False
+    post_install = True
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestSaleOrderDates, cls).setUpClass()
+        cls.env = cls.env(
+            context=dict(cls.env.context, tracking_disable=True, tz="UTC")
+        )
+        cls.setupClassCompany()
+        cls.setUpClassCalendar()
+        cls.setUpClassWarehouse()
+        cls.setUpClassPartner()
+        cls.setUpClassProduct()
+
+    @classmethod
+    def _define_calendar(cls, name, attendances):
+        return cls.env["resource.calendar"].create(
+            {
+                "name": name,
+                "attendance_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "%s_%d" % (name, index),
+                            "hour_from": att[0],
+                            "hour_to": att[1],
+                            "dayofweek": str(att[2]),
+                        },
+                    )
+                    for index, att in enumerate(attendances)
+                ],
+            }
+        )
+
+    @classmethod
+    def setupClassCompany(cls):
+        cls.company = cls.env.user.company_id
+        cls.company.security_lead = 1
+
+    @classmethod
+    def setUpClassCalendar(cls):
+        cls.calendar = cls._define_calendar("40 Hours", [(8, 16, i) for i in range(5)],)
+
+    @classmethod
+    def setUpClassWarehouse(cls):
+        cls.warehouse = cls.env.ref("stock.warehouse0")
+        cls.warehouse.write(
+            {
+                "cutoff_time": CUTOFF_TIME,
+                "apply_cutoff": True,
+                "calendar_id": cls.calendar,
+            }
+        )
+
+    @classmethod
+    def setUpClassPartner(cls):
+        cls.customer = cls.env.ref("base.res_partner_12")
+        cls.customer.order_delivery_cutoff_preference = "warehouse_cutoff"
+        cls.customer.delivery_time_preference = "workdays"
+
+    @classmethod
+    def setUpClassProduct(cls):
+        cls.product = cls.env.ref("product.product_product_3")
+
+    def _create_order(self, dt):
+        with freeze_time(dt.strftime("%Y-%m-%d %H:%M:%S")):
+            order = self.env["sale.order"].create(
+                {"partner_id": self.customer.id, "warehouse_id": self.warehouse.id}
+            )
+            self.env["sale.order.line"].create(
+                {
+                    "order_id": order.id,
+                    "product_id": self.product.id,
+                    "product_uom_qty": 1,
+                    "customer_lead": 1,
+                }
+            )
+            return order
+
+    @freeze_time(FRIDAY_BEFORE_CUTOFF)
```

Comment:
You should make an order on Saturday before cutoff and you should see this test failing without the fix I mentioned